lib/gollum-lib/page.rb

```diff
@@ -389,10 +382,10 @@ def find_page_in_tree(map, name, checked_dir = nil, exact = false)
       name        = ::File.join(checked_dir, name) if checked_dir
 
       map.each do |entry|
-        next if entry.name.to_s.empty?
-        entry_name = Page.canonicalize_filename(entry.name)
+        next if entry.name.to_s.empty? || !self.class.valid_extension?(entry.name)
+        entry_name =  ::File.extname(name).empty? ? Page.canonicalize_filename(entry.name) : entry.name
         path = checked_dir ? ::File.join(entry.dir, entry_name) : entry_name
-        next unless name == path
+        next unless name.downcase == path.downcase
```

Comment:
I think your snippet with the bang methods is right for now. When ruby 3.0 rolls around we'll have to go through the whole code base anyway.